nifi-stateless/nifi-stateless-bootstrap/src/main/java/org/apache/nifi/stateless/bootstrap/AllowListClassLoader.java

```diff
@@ -37,36 +43,92 @@
  * </p>
  */
 public class AllowListClassLoader extends ClassLoader {
-    private final Set<String> allowed;
+    private static final Logger logger = LoggerFactory.getLogger(AllowListClassLoader.class);
+
+    private final Set<String> allowedClassNames;
+    private final List<String> allowedModulePrefixes = Arrays.asList("java.", "jdk.");
```

Comment:
I think the same module check would work on Java 9+ JDK too so it may make sense to use it instead of loading the classes from `jmod` files and to have a unified approach for Java 9+ JDK/JRE.

Reply:
Yeah, good point.